docs/learning/tutorials/hello-world-viewer.md

```diff
@@ -0,0 +1,288 @@
+
+Our ```helloWidget``` consists of three attributes:
+
+1. ``` id ``` - used to uniquely identify the widget
+2. ``` label ``` - description label for our widget
+3. ``` getWidgetContent() ``` - returns our custom UI component
+
+At this point we need to import ```MyFirstUiProvider``` at the top of file ```App.tsx```:
+
+``` Typescript
+import { MyFirstUiProvider } from "./MyFirstUiProvider";
+```
+
+Finally, let's clean up the ```span``` and ```div``` that we added directly into the ```App``` component earlier.  Now the ``` return ``` statement in ```App.tsx``` should look like this:
+
+``` typescript
+  return (
+    <div className="viewer-container">
+      <Header
+        handleLogin={onLoginClick}
+        loggedIn={isAuthorized}
+        handleLogout={onLogoutClick}
+      />
+      {isLoggingIn ? (
+        <span>"Logging in...."</span>
+      ) : (
+        isAuthorized && (
+          <Viewer
+            contextId={process.env.IMJS_CONTEXT_ID ?? ""}
+            iModelId={process.env.IMJS_IMODEL_ID ?? ""}
+            authConfig={{ oidcClient: AuthorizationClient.oidcClient }}
+            uiProviders={[new MyFirstUiProvider()]}
+          />
+        )
+      )}
+    </div>
+  );
+```
+
+Now we have our "Hello World" ```span``` displaying in a panel within the ```Viewer``` component.  It should look like this:
+
+![HelloWorldWidget]($docs/learning/tutorials/images/hello_world_widget.png)
+
+### Beyond Hello World
+
+Saying hello to the world can be fun but we need to get past that.  For this next step we'll swap out our trivial ```helloWidget``` with something a little more interactive: a [Toggle]($docs/reference/ui-core/toggle/toggle/).  Eventually this toggle will control the background color, so we'll name our new widget ```backgroundColorWidget```.  Instead of returning a ```span``` we'll return a ```Toggle```.
+
+Start by navigating back to ```MyFirstUiProvider.tsx``` and adding an import for ```Toggle``` at the top of the file:
+
+``` typescript
+import { Toggle } from "@bentley/ui-core";
+```
+
+Next switch out the ```helloWidget``` with the new ```backgroundColorWidget``` here:
+
+``` typescript
+    if (location === StagePanelLocation.Right) {
+      const backgroundColorWidget: AbstractWidgetProps = {
+        id: "BackgroundColorWidget",
+        label: "Background Color Toggle",
+        getWidgetContent() {
+          return <Toggle></Toggle>
```

Comment:
nit: shouldn't need the close tag for the toggle

```suggestion
          return <Toggle />
```

Reply:
Fixed

Reply:
Could you update the rest of the file to remove the toggle closing tag too? I know its a huge nit pick... sorry!

Reply:
No worries, Fixed.